app/presenters/sufia/work_show_presenter.rb

```diff
@@ -42,6 +42,10 @@ def stats_path
       Sufia::Engine.routes.url_helpers.stats_work_path(self)
     end
 
+    def member_presenters(ids = ordered_ids, presenter_class = composite_presenter_class)
+      super.delete_if { |presenter| current_ability.cannot?(:read, presenter.solr_document) }
```

Comment:
@jcoyne Sorry, neglected to say that this is the start of a fix for #2394.   I'm using the fix already in ScholarSphere as a start https://github.com/psu-stewardship/scholarsphere/commit/206f1738ff6051b4634ee04f5a18fbe9310fd4b5